build_web_compilers/lib/src/dev_compiler_bootstrap.dart

```diff
@@ -321,6 +322,36 @@ require.config({
 });
 ''';
 
+/// Hot-reload config
+///
+/// Listen WebSocket for updates in build results
+///
+/// Now only ilve-reload functional - just reload page on update message
+final _hotReloadConfig = '''
+(function() {
+  var wsUrl; 
+  if (baseUrl.startsWith('http')) {
+    wsUrl = baseUrl.replace('http', 'ws');
+  } else {
+    // TODO: when it may make sense? WebWorkers? Tests?
+    wsUrl = 'ws://' + location.host + baseUrl;
+  }
+  wsUrl += '\$hotreload';
+  try {
+    var ws = new WebSocket(wsUrl);
+    ws.onmessage = function(event) {
+      console.log(event);
+      if(event.data === 'update'){
+        location.reload();
+      }
+    };
+  } catch (e) {
+    // Live reload might be turned off. Just ignore it for now
+    // TODO: Find a way to pass serve option here, to not inject this code at all, if live/hot reload is turned off
```

Comment:
This is an interesting problem - ideally this package does not have to rely on the implementation details in the server (such as the `update` event, or the `$hotreload` path). We want to keep breaking changes in build_runner to a minimum, so ideally we wouldn't have other packages depending on these details. 

To that end, it might make sense to actually inject this code from the server itself, possibly into all `html` requests that it serves, although that could be challenging as well. It would be worth talking to @grouma about how we do this for DDR internally.

Reply:
It looks like non-trivial task to figure out if request is html page or not. Both `Content-type` header and `<html>` tag might be omitted.
Maybe it's easier to inject code to the end of any `*.dart.bootstrap.js` file?

Reply:
> Maybe it's easier to inject code to the end of any *.dart.bootstrap.js file?

The problem with that is we are then backing knowledge of build_web_compilers into build_runner - which is maybe worse than the other way around haha.

Reply:
DDR knows ahead of time what the main entry JS file is called. It then intercepts this request and returns the DDR application. This application will embed a request for the main entry along with a request parameter `forReal`. The logic can be found in `ddr_handler.dart`. This pattern has been helpful because we have completely changed our injected application logic a number of times. Is it possible to know what the root request will be in build_runner?